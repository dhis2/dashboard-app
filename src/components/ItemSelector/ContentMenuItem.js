import React from 'react';
import { MenuItem, colors } from '@dhis2/ui-core';
import i18n from '@dhis2/d2-i18n';

import { getItemIcon } from '../../modules/itemTypes';
import LaunchIcon from '../../icons/Launch';

import classes from './styles/ContentMenuItem.module.css';

const LaunchLink = ({ url }) => (
    <a
        onClick={e => e.stopPropagation()}
        className={classes.launchLink}
        target="_blank"
        rel="noopener noreferrer"
        href={url}
    >
        <LaunchIcon />
    </a>
);

const InsertButton = () => (
    <button className={classes.buttonInsert}>{i18n.t('Insert')}</button>
);

const ContentMenuItem = ({ type, name, onInsert, url }) => {
    const ItemIcon = getItemIcon(type);

    return (
        <MenuItem
            dense
            onClick={onInsert}
            label={
                <div className={classes.menuItem}>
                    <div className={classes.label}>
                        <ItemIcon
                            style={{ margin: '6px', fill: colors.grey600 }}
                        />
                        <span>{name}</span>
                        {url ? <LaunchLink url={url} /> : null}
                    </div>
                    <InsertButton />
                </div>
            }
        />
    );
};

export default ContentMenuItem;
