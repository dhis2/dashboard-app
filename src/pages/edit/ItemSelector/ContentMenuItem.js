import { visTypeIcons } from '@dhis2/analytics'
import i18n from '@dhis2/d2-i18n'
import { MenuItem, colors, IconLaunch16 } from '@dhis2/ui'
import PropTypes from 'prop-types'
import React from 'react'
import { getItemIcon, VISUALIZATION } from '../../../modules/itemTypes.js'
import classes from './styles/ContentMenuItem.module.css'

const LaunchLink = ({ url }) => (
    <a
        onClick={(e) => e.stopPropagation()}
        className={classes.launchLink}
        target="_blank"
        rel="noopener noreferrer"
        href={url}
    >
        <IconLaunch16 color={colors.grey700} />
    </a>
)

LaunchLink.propTypes = {
    url: PropTypes.string,
}

const InsertButton = () => (
    <button className={classes.buttonInsert}>{i18n.t('Insert')}</button>
)

const ContentMenuItem = ({ type, name, onInsert, url, visType }) => {
    const ItemIcon = getItemIcon(type)

    let renderedItemIcon
    if (type === VISUALIZATION) {
        const VisualizationIcon = visTypeIcons[visType]
        renderedItemIcon = <VisualizationIcon color={colors.grey600} />
    } else {
        renderedItemIcon = <ItemIcon color={colors.grey600} />
    }

    return (
        <MenuItem
            onClick={onInsert}
            icon={renderedItemIcon}
            label={
                <div className={classes.menuItem}>
                    <div className={classes.label}>
                        <span>{name}</span>
                        {url ? <LaunchLink url={url} /> : null}
                    </div>
                    <InsertButton />
                </div>
            }
            dataTest={`menu-item-${name}`}
        />
    )
}

ContentMenuItem.propTypes = {
    name: PropTypes.string,
    type: PropTypes.string,
    url: PropTypes.string,
    visType: PropTypes.string,
    onInsert: PropTypes.func,
}

export default ContentMenuItem
