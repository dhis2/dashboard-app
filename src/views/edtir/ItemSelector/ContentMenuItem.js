import React from 'react'
import { MenuItem, colors } from '@dhis2/ui'
import i18n from '@dhis2/d2-i18n'
import PropTypes from 'prop-types'
import { visTypeIcons, VIS_TYPE_SCATTER } from '@dhis2/analytics'

import ScatterIcon from './assets/ScatterIcon'
import LaunchIcon from './assets/Launch'
import { getItemIcon, VISUALIZATION } from '../../../modules/itemTypes'

import classes from './styles/ContentMenuItem.module.css'

const LaunchLink = ({ url }) => (
    <a
        onClick={e => e.stopPropagation()}
        className={classes.launchLink}
        target="_blank"
        rel="noopener noreferrer"
        href={url}
    >
        <LaunchIcon />
    </a>
)

LaunchLink.propTypes = {
    url: PropTypes.string,
}

const InsertButton = () => (
    <button className={classes.buttonInsert}>{i18n.t('Insert')}</button>
)

const ContentMenuItem = ({ type, name, onInsert, url, visType }) => {
    const ItemIcon = getItemIcon(type)

    let renderedItemIcon
    if (type === VISUALIZATION) {
        renderedItemIcon =
            visType === VIS_TYPE_SCATTER ? (
                <ScatterIcon />
            ) : (
                visTypeIcons[visType]
            )
    } else {
        renderedItemIcon = <ItemIcon style={{ fill: colors.grey600 }} />
    }

    return (
        <MenuItem
            dense
            onClick={onInsert}
            label={
                <div className={classes.menuItem}>
                    <div className={classes.label}>
                        <span style={{ marginRight: '6px' }}>
                            {renderedItemIcon}
                        </span>
                        <span>{name}</span>
                        {url ? <LaunchLink url={url} /> : null}
                    </div>
                    <InsertButton />
                </div>
            }
            dataTest={`menu-item-${name}`}
        />
    )
}

ContentMenuItem.propTypes = {
    name: PropTypes.string,
    type: PropTypes.string,
    url: PropTypes.string,
    visType: PropTypes.string,
    onInsert: PropTypes.func,
}

export default ContentMenuItem
