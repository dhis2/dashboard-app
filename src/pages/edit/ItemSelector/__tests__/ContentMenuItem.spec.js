import { render } from '@testing-library/react'
import React from 'react'
import ContentMenuItem from '../ContentMenuItem'

test('ContentMenuItem has a LaunchLink when url is provided', () => {
    const props = {
        name: 'Rainbow Dash',
        type: 'pony',
        url: 'http://ponies-r-us.com',
        visType: 'BAR',
        onInsert: jest.fn(),
    }

    const { container } = render(<ContentMenuItem {...props} />)

    expect(container).toMatchSnapshot()
})

test('does not have LaunchLink if no url provided', () => {
    const props = {
        name: 'Fancy chart',
        type: 'VISUALIZATION',
        url: '',
        visType: 'BAR',
        onInsert: jest.fn(),
    }
    const { container } = render(<ContentMenuItem {...props} />)
    expect(container).toMatchSnapshot()
})
