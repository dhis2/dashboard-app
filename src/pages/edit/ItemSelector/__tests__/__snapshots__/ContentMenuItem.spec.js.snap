// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentMenuItem has a LaunchLink when url is provided 1`] = `
<div>
  <li
    class="jsx-665727467 "
    data-test="menu-item-Rainbow Dash"
  >
    <a
      class="jsx-665727467"
    >
      <span
        class="jsx-665727467 icon"
      >
        <svg
          color="#6e7a8a"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm0 12a1 1 0 110 2 1 1 0 010-2zm0-10a4 4 0 011.155 7.83l-.155.043V15h-2v-3h1a2 2 0 10-1.995-2.15L10 10H8a4 4 0 014-4z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="jsx-665727467 label"
      >
        <div
          class="menuItem"
        >
          <div
            class="label"
          >
            <span>
              Rainbow Dash
            </span>
            <a
              class="launchLink"
              href="http://ponies-r-us.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                color="#4a5768"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 2v1H3v10h10V9h1v4a1 1 0 01-1 1H3a1 1 0 01-1-1V3a1 1 0 011-1zm7.5-1a.5.5 0 01.492.41L15 1.5v5a.5.5 0 01-.992.09L14 6.5V2.706L7.354 9.354a.5.5 0 01-.765-.638l.057-.07L13.292 2H9.5a.5.5 0 01-.492-.41L9 1.5a.5.5 0 01.41-.492L9.5 1z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
          <button
            class="buttonInsert"
          >
            Insert
          </button>
        </div>
      </span>
    </a>
  </li>
</div>
`;

exports[`does not have LaunchLink if no url provided 1`] = `
<div>
  <li
    class="jsx-665727467 "
    data-test="menu-item-Fancy chart"
  >
    <a
      class="jsx-665727467"
    >
      <span
        class="jsx-665727467 icon"
      >
        <svg
          color="#6e7a8a"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 2v18h18v2H2V2h2zm16 13v3H6v-3h14zm-4-5v3H6v-3h10zm-4-5v3H6V5h6z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="jsx-665727467 label"
      >
        <div
          class="menuItem"
        >
          <div
            class="label"
          >
            <span>
              Fancy chart
            </span>
          </div>
          <button
            class="buttonInsert"
          >
            Insert
          </button>
        </div>
      </span>
    </a>
  </li>
</div>
`;
